packages/openapi-ts/src/plugins/@hey-api/client-core/bundle/utils.ts

```diff
@@ -110,3 +110,31 @@ export const getUrl = ({
   }
   return url;
 };
+
+export function getValidRequestBody(options: {
+  body?: unknown;
+  bodySerializer?: BodySerializer | null;
+  serializedBody?: unknown;
+}) {
+  const hasBody = options.body !== undefined;
+  const isSerializedBody = hasBody && options.bodySerializer;
+
+  if (isSerializedBody && Object.hasOwn(options, 'serializedBody')) {
```

Comment:
Maybe more succinct, it shouldn't make a difference here

```suggestion
  if (isSerializedBody && 'serializedBody' in options) {
```

You could also move the condition below inside this block so we check `isSerializedBody` only once

Reply:
i can update. I avoided the nested ifs but not a big deal in this case.